wandb/sdk/wandb_init.py

```diff
@@ -312,6 +312,9 @@ def _log_setup(self, settings):
         logger.info("Logging internal logs to {}".format(settings.log_internal))
 
     def _atexit_cleanup(self):
+        if self.backend is None:
+            logger.warning("process exited without backend configured")
+            return False
```

Comment:
Was only seeing this in CI, not sure how we could hookup the exit func and not have a backend, but added a check here.  Not sure if it's right.

Reply:
seems fine.. atexit needs hardening.